service/history/config/config_test.go

```diff
@@ -0,0 +1,356 @@
+		"EnableRetryForChecksumFailure":                        {dynamicconfig.EnableRetryForChecksumFailure, true},
+		"EnableHistoryCorruptionCheck":                         {dynamicconfig.EnableHistoryCorruptionCheck, true},
+		"NotifyFailoverMarkerInterval":                         {dynamicconfig.NotifyFailoverMarkerInterval, time.Second},
+		"NotifyFailoverMarkerTimerJitterCoefficient":           {dynamicconfig.NotifyFailoverMarkerTimerJitterCoefficient, 1.0},
+		"EnableGracefulFailover":                               {dynamicconfig.EnableGracefulFailover, true},
+		"EnableActivityLocalDispatchByDomain":                  {dynamicconfig.EnableActivityLocalDispatchByDomain, true},
+		"MaxActivityCountDispatchByDomain":                     {dynamicconfig.MaxActivityCountDispatchByDomain, 100},
+		"ActivityMaxScheduleToStartTimeoutForRetry":            {dynamicconfig.ActivityMaxScheduleToStartTimeoutForRetry, time.Second},
+		"EnableDebugMode":                                      {dynamicconfig.EnableDebugMode, true},
+		"EnableTaskInfoLogByDomainID":                          {dynamicconfig.HistoryEnableTaskInfoLogByDomainID, true},
+		"EnableTimerDebugLogByDomainID":                        {dynamicconfig.EnableTimerDebugLogByDomainID, true},
+		"SampleLoggingRate":                                    {dynamicconfig.SampleLoggingRate, 100},
+		"EnableShardIDMetrics":                                 {dynamicconfig.EnableShardIDMetrics, true},
+		"LargeShardHistorySizeMetricThreshold":                 {dynamicconfig.LargeShardHistorySizeMetricThreshold, 100},
+		"LargeShardHistoryEventMetricThreshold":                {dynamicconfig.LargeShardHistoryEventMetricThreshold, 100},
+		"LargeShardHistoryBlobMetricThreshold":                 {dynamicconfig.LargeShardHistoryBlobMetricThreshold, 100},
+		"EnableStrongIdempotency":                              {dynamicconfig.EnableStrongIdempotency, true},
+		"EnableStrongIdempotencySanityCheck":                   {dynamicconfig.EnableStrongIdempotencySanityCheck, true},
+		"GlobalRatelimiterNewDataWeight":                       {dynamicconfig.HistoryGlobalRatelimiterNewDataWeight, 1.0},
+		"GlobalRatelimiterUpdateInterval":                      {dynamicconfig.GlobalRatelimiterUpdateInterval, time.Second},
+		"GlobalRatelimiterDecayAfter":                          {dynamicconfig.HistoryGlobalRatelimiterDecayAfter, time.Second},
+		"GlobalRatelimiterGCAfter":                             {dynamicconfig.HistoryGlobalRatelimiterGCAfter, time.Second},
+		"HostName":                                             {nil, hostname},
+	}
+	client := dynamicconfig.NewInMemoryClient()
+	for fieldName, expected := range fields {
+		if expected.key != nil {
+			err := client.UpdateValue(expected.key, expected.value)
+			if err != nil {
+				t.Errorf("Failed to update config for %s: %s", fieldName, err)
+			}
+		}
+	}
+	dc := dynamicconfig.NewCollection(client, testlogger.New(t))
```

Comment:
I kinda wonder if this would be better to do as a "pick a hashed value for whatever key was requested"... the keys and values sorta don't matter, just that they're predictable and come from the dynamic config.

Reply:
the values have types. And the type matters.